documentation/modules/auxiliary/scanner/http/http_hsts.md

```diff
@@ -0,0 +1,66 @@
+## Vulnerable Application
+
+### Description
+
+This module checks to see whether or not the scanned systems return the HSTS header to enforce HSTS.
+
+## Verification Steps
+
+1. Do: ```use auxiliary/scanner/http/http_hsts```
+2. Do: ```set RHOSTS [IP]```
+3. Do: ```set RPORT [PORT]```
+4. Do: ```run```
+
+## Options
+
+  **Proxies**
+
+  Any proxies to be used, in the format format type:host:port[,type:host:port][...]. Default is `` or none.
+
+  **RHOSTS**
+
+  The target hosts to be scanned. Can be a single host, a range CIDR identifier, hosts file with syntax `file:<path>`. Default is ``.
+
+  **RPORT**
+
+  Port to scan on the targets.  Default is `443`.
+
+  **SSL**
+
+  Should SSL be used when connecting? Default is `TRUE`.
+
+  **THREADS**
+  Number of threads to use to complete the scan. Default is `1`.
+
+  **VHOST**
+  Virtual host of the HTTP servers being scanned. Default is ``.
+
+## Scenarios
+
+**Running the scanner**
+
+```
+msf5 > use auxiliary/scanner/http/http_hsts
+msf5 auxiliary(scanner/http/http_hsts) > show options
+
+Module options (auxiliary/scanner/http/http_hsts):
+
+   Name     Current Setting  Required  Description
+   ----     ---------------  --------  -----------
+   Proxies                   no        A proxy chain of format type:host:port[,type:host:port][...]
+   RHOSTS                    yes       The target host(s), range CIDR identifier, or hosts file with syntax 'file:<path>'
+   RPORT    443              yes       The target port (TCP)
+   SSL      true             no        Negotiate SSL/TLS for outgoing connections
+   THREADS  1                yes       The number of concurrent threads (max one per host)
+   VHOST                     no        HTTP server virtual host
+
+msf5 auxiliary(scanner/http/http_hsts) > set RHOSTS 192.168.24.20
+RHOSTS => 192.168.24.20
+msf5 auxiliary(scanner/http/http_hsts) > run
+
+[+] 192.168.24.20:443 - Strict-Transport-Security:max-age=31536000; includeSubDomains
+[*] Scanned 1 of 1 hosts (100% complete)
+[*] Auxiliary module execution completed
+msf5 auxiliary(scanner/http/http_hsts) >
+
+```
```

Comment:
If there is a tool that can verify this other than metasploit, I like to include it as well.

Something like https://github.com/rapid7/metasploit-framework/pull/12450/files#diff-b2c1aa63724b140259932dfb390a0711R356

https://nmap.org/nsedoc/scripts/http-security-headers.html

Only, I think it should be an H3 (@tperry-r7 agree?)

Reply:
Should I try run another tool and include the output as a H3 header here? I know Nmap should have a NSE script that will provide similar information so can include that output if needed.

Reply:
Sorry ignore my last comment, I see what you mean now :)

Reply:
I usually just have 1 other tool if its free and on Kali (aka common).  It helps for when youre on a test and one tool says one thing, a 2nd tool says something else, and you need a tie breaker.  Not a requirement, but honestly `nmap -p 80 -sV --script <abc> <ip>` won't take but 3min to add.

Reply:
Next commit should have the output from running the http-security-headers script in Nmap on the Apache host that I set up.